src/event.rs

```diff
@@ -1,28 +1,32 @@
-use crate::context::Context;
-use crate::AResult;
+use crate::{
+    context::{Context, Frame, SFrame},
+    AResult,
+};
+use kafka::{
+    error::Error as KafkaError,
+    producer::{Producer, Record, RequiredAcks},
+};
 use serde;
 use serde_json;
-use std::time::Instant;
+use std::{
+    fs::{create_dir, File},
+    io::Write,
+    path::Path,
+    time::Duration,
+    time::Instant,
+};
 
-pub trait Event<T> {
-    fn map(&self, ctx: &Context, _data: &T) -> AResult<serde_json::Value>;
+pub trait Event<T, K> {
+    fn map(&self, ctx: &Context, _data: &AResult<T>) -> AResult<K>;
 
-    fn with<F>(&self, ctx: &Context, cb: F) -> AResult<T>
-    where
-        F: FnOnce() -> T,
-    {
-        let start = Instant::now();
-        let r = cb();
-        let start2 = Instant::now();
-        let result = self.map(ctx, &r)?;
-        println!(
-            "result: {:?}\ncb_time: {:?}\nmap_time: {:?}",
-            result,
-            start2.duration_since(start),
-            Instant::now().duration_since(start2)
-        );
-        Ok(r)
+    fn name(&self) -> String;
+
+    fn destination(&self) -> String;
```

Comment:
Not fond of either name or destination now. I think we discussed, but lets get rid of it. If needed we can use some macro to auto generate the name based on module path of the Event struct.